proofread: for_distributed.md (#1589)
juicedata/juicefs: docs/en/getting-started/for_distributed.md

# JuiceFS Quick Start Guide for Distributed Mode

The previous document ["JuiceFS Quick Start Guide for Standalone Mode "](for_local.md) introduces how to create a file system that can be mounted on any host by using an "object storage" and a "SQLite" database. Thanks to the feature that the object storage is accessible by any computer with privileges on the network, we can also access the same JuiceFS file system on different computers by simply copying the SQLite database file to any computer that needs to access the storage.

> However, the real-time availability of the files is not guaranteed if the file system is shared by the above approach. Since SQLite is a single file database that cannot be accessed by multiple computers at the same time, the database that supports network access is needed, such as Redis, PostgreSQL, MySQL, etc., which allows a file system to be mounted and read by multiple computers in a distributed environment. 

Author: ```suggestion
However, the real-time availability of the files is not guaranteed if the file system is shared by the above approach. Since SQLite is a single file database that cannot be accessed by multiple computers at the same time, a database that supports network access is needed, such as Redis, PostgreSQL, MySQL, etc., which allows a file system to be mounted and read by multiple computers in a distributed environment.
```

In this document, a multi-user "cloud database" is used to replace the single-user "SQLite" database used in the previous document, aiming to perform the distributed file system that can be mounted on any computer on the network for reading and writing.

## Network Database

The meaning of "Network Database" here refers to the database that allows multiple users to access it simultaneously through the network. From this perspective, the database can be simply divided into:

> 1. **Standalone Database**: which is single file and usually only accessible by a single machine, such as SQLite, Microsoft Access, etc.

Author: ```suggestion
1. **Standalone Database**: which is a single-file database and is usually only accessed locally, such as SQLite, Microsoft Access, etc.
```

```

:::note
By default, CentOS 6 does not mount the network file system during boot, you need to run the command `sudo chkconfig --add netfs` to enable automatically mounting.
:::

### 6. Unmounting the file system

You can unmount the JuiceFS file system (assuming the mount point path is `mnt`) by the command `juicefs umount`.

```shell
juicefs umount mnt
```

#### Unmounting failure

If the command fails to unmount the file system after execution, it will prompt `Device or resource busy`.

```shell
2021-05-09 22:42:55.757097 I | fusermount: failed to unmount mnt: Device or resource busy
exit status 1
```

> This failure happens probrobly because some programs are reading and writing files in the file system when the unmount command is executed. To ensure data security, you should first troubleshoot which programs are interacting with files on the file system (e.g. via the command `lsof`) and try to terminate the interaction between them before re-executing the unmount command.

Author: This failure happens probably because some programs are reading or writing files in the file system when executing `unmount` command. To avoid data loss, you should first find out which processes are accessing files in the file system (e.g. via the command `lsof`) and try to release the files before re-executing the `unmount` command.

:::caution
The following command may result in file corruption and loss, so be careful to use it!
:::

> Once you have ensured data security, add the option `--force` or `-f` to the unmount command to force the file system unmounted.

Author: ```suggestion
You can add the option `--force` or `-f` to force the file system unmounted if the data safety is ensured. 
```

> Once you have ensured data security, add the option `--force` or `-f` to the unmount command to force the file system unmounted.

Author: ```suggestion
You can add the option `--force` or `-f` to force the file system unmounted if you are pretty sure what you are doing.
```
